### PR TITLE
Wrap getContainerRenderWindow loop in try/catch

### LINKED ISSUE #798: Bug: A security error occurred and caused a crash when used in a different source iframe
https://github.com/casesandberg/react-color/blob/bc9a0e1dc5d11b06c511a8e02a95bd85c7129f4b/src/components/common/Saturation.js#L23

The lack of strict access to parent here leads to security errors
Please fix it in time

image

### PR DESCRIPTION
**Pull request by itsjoekent**
Fixes https://github.com/casesandberg/react-color/issues/798

@casesandberg Can you merge + publish this? This is causing our application to fail in production

### CODE REVIEW

File: src/components/common/Saturation.js [hunk 1]
```diff
@@ -20,9 +20,11 @@ export class Saturation extends (PureComponent || Component) {
...
     const { container } = this
     let renderWindow = window
-    while (!renderWindow.document.contains(container) && renderWindow.parent !== renderWindow) {
-      renderWindow = renderWindow.parent
-    }
+    try {
+      while (!renderWindow.document.contains(container) && renderWindow.parent !== renderWindow) {
+        renderWindow = renderWindow.parent
+      }
+    } catch (error) {}
```
Comment (lixun910) [inline src/components/common/Saturation.js:27]: @casesandberg @itsjoekent For cross-domain error, the `renderWindow.document` will raise DOMException. At this point, is the renderWindow already an invalid object, which can not be returned?

This works on my test:
```javascript
    let renderWindow = window
    let lastRenderWindow = window
    try {
        while (!renderWindow.document.contains(container) && renderWindow.parent !== renderWindow) {
          lastRenderWindow = renderWindow
          renderWindow = renderWindow.parent
        }
      } catch(error) {
        // handle possible cross-origin error, if happens then use current window in frame
      }
      return lastRenderWindow
```

#806

Comment (zhouyashu279) [reply src/components/common/Saturation.js:27]: For cross-domain error , it's cannot access renderWindow.document   when renderWindow is parent iframe

Comment (zhouyashu279) [reply src/components/common/Saturation.js:27]: issue #806 , Request resolution

Comment (zhouyashu279) [reply src/components/common/Saturation.js:27]: > 

use it in inner iframe, when  cross-domain, 'renderWindow.document' will throw error, please reconfirm and repair

**Comment by jdsxzhao** [comment]
When can this pr be merged? The problem mentioned in the issue https://github.com/casesandberg/react-color/issues/806 is still causing crashes!

**Pull request closed without merging**